### PR TITLE
Add kube-system as an immortal namespace

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

File: pkg/cmd/server/origin/admission/chain_builder.go [hunk 1]
```diff
@@ -198,7 +198,7 @@ func newAdmissionChain(pluginNames []string, admissionConfigFilename string, opt
 		switch pluginName {
 		case lifecycle.PluginName:
 			// We need to include our infrastructure and shared resource namespaces in the immortal namespaces list
-			immortalNamespaces := sets.NewString(metav1.NamespaceDefault)
+			immortalNamespaces := sets.NewString(metav1.NamespaceDefault, metav1.NamespaceSystem)
```
Comment (deads2k) [inline pkg/cmd/server/origin/admission/chain_builder.go]: This still isn't in sync with upstream.  They have `NamespacePublic` protected too.

**Comment by deads2k** [comment]
opened https://github.com/openshift/origin/pull/17914 as an alternative in head.

File: pkg/cmd/server/origin/admission/chain_builder.go (hunk 1 above)
Comment (liggitt) [reply pkg/cmd/server/origin/admission/chain_builder.go]: I don't actually care about public… I care about the namespaces which, when deleted, require cluster-admin intervention

Comment (liggitt) [reply pkg/cmd/server/origin/admission/chain_builder.go]: I don't really object to public either

**Comment by liggitt** [comment]
closing in favor of https://github.com/openshift/origin/pull/17914

**Pull request closed without merging**